serenity/serenity.hpp

```diff
@@ -26,6 +26,41 @@ class Consumer : public BusSocket {
 };
 
 
+/**
+ * This consumer requires that all producents will run consume on
+ * this instance. (Even during error flow).
+ */
+template<typename T>
+class MultiConsumer : public Consumer<T> {
+ public:
+  explicit MultiConsumer(uint64_t _producentsToWaitFor)
+    : producentsToWaitFor(_producentsToWaitFor) {
+    CHECK_ERR(_producentsToWaitFor > 0);
+  }
+
+  virtual ~MultiConsumer() {}
+
+  Try<Nothing> consume(const T& in) {
+    this->products.push_back(in);
+
+    if (this->products.size() == this->producentsToWaitFor) {
+      // Run virtual function which should be implemented in derived
+      // class.
+      this->_consume(this->products);
```

Comment:
We can do multi consumption without this class.
This class is about synchronization,